odoo/tools/translate.py

```diff
@@ -886,107 +914,152 @@ def push_translation(module, type, name, id, source, comments=None):
-        query += ' WHERE module IN ( SELECT name FROM ir_module_module WHERE state = \'installed\') '
+        A record is considered as untranslatable if:
+        - it does not exist
+        - the model is flagged with _translate=False
+        - it is a field of a model flagged with _translate=False
+        - it is a selection of a field of a model flagged with _translate=False
 
-    if 'all' not in modules:
-        query += ' WHERE module IN %s'
-        query_param = (tuple(modules),)
-    else:
-        query += ' WHERE module != %s'
-        query_param = ('__export__',)
+        :param records: a list of namedtuple belonging to the same model
+        """
+        model = next(iter(records)).model
+        if model not in self.env:
+            _logger.error("Unable to find object %r", model)
+            return False
 
-    query += ' GROUP BY model, res_id, module ORDER BY module, model, min(name)'
+        if not self.env[model]._translate:
+            return False
 
-    cr.execute(query, query_param)
 
-    for (xml_name, model, res_id, module) in cr.fetchall():
-        xml_name = "%s.%s" % (module, xml_name)
+        res_ids = [r.res_id for r in records]
+        records = self.env[model].browse(res_ids).exists()
+        if len(records) < len(res_ids):
+            missing_ids = set(records.ids) - set(res_ids)
+            _logger.warning("Unable to find objects %r with id %d", model, ', '.join(missing_ids))
+            if not records:
+                return False
 
-        if model not in env:
-            _logger.error(u"Unable to find object %r", model)
-            continue
+        if model == 'ir.model.fields.selection':
+            fields = defaultdict(list)
+            for selection in records:
+                fields[selection.field_id] = selection
+            for field, selection in fields.items():
+                field_name = field.name
+                field_model = self.env.get(field.model)
+                if (field_model is None or not field_model._translate or
+                        field_name not in field_model._fields):
+                    # the selection is linked to a model with _translate=False, remove it
+                    records -= selection
+        elif model == 'ir.model.fields':
+            for field in records:
+                field_name = field.name
+                field_model = self.env.get(field.model)
+                if (field_model is None or not field_model._translate or
+                        field_name not in field_model._fields):
+                    # the field is linked to a model with _translate=False, remove it
+                    records -= field
+
+        return records
+
+
+    def _export_translatable_records(self):
+        """ Export translations of all translated records having an external id """
+
+        query = """SELECT min(name), model, res_id, module
+                     FROM ir_model_data
+                    WHERE {module_clause}
```

Comment:
It is always the same `module_clause`, so
```suggestion
                    WHERE module = ANY(%s)
```